readthedocs/projects/admin.py

```diff
@@ -177,6 +185,40 @@ def delete_selected_and_artifacts(self, request, queryset):
                 broadcast(type='app', task=remove_dir, args=[project.doc_path])
         return delete_selected(self, request, queryset)
 
+    def mark_as_abandoned(self, request, queryset):
+        """
+        Marks selected projects as abandoned.
+
+        It adds '-abandoned' to the slug, marks
+        Project.is_abandoned and sends an email to the user
+        notifying the change and link to the new project/docs
+        page.
+        """
+        queryset.update(
+            slug=Concat('slug', Value('-abandoned')),
+            is_abandoned=True
+        )
+
+    mark_as_abandoned.short_description = 'Mark as abandoned'
+
+    def request_namespace(self, request, queryset):
+        """Notify user that their project's namespace has been requested via email"""
+        qs_iterator = queryset.iterator()
+        for project in qs_iterator:
+            users = project.users.get_queryset()
+            for user in users:
+                send_email(
+                    recipient=user.email,
```

Comment:
I am a little confused on this one.
Should the notification be sent to every user?

Reply:
I think so.

These users are the maintainers of the project, so I think it's correct to send emails to them.

Reply:
ok, I get it.
Thanks